deps/build.jl

```diff
@@ -2,53 +2,75 @@
     using Libdl
 end
 
-depsfile = joinpath(dirname(@__FILE__),"deps.jl")
+depsfile = joinpath(dirname(@__FILE__), "deps.jl")
 if isfile(depsfile)
     rm(depsfile)
 end
 
 function write_depsfile(path)
-    open(depsfile,"w") do f
-        print(f,"const libcplex = ")
+    open(depsfile, "w") do f
+        print(f, "const libcplex = ")
         show(f, path) # print with backslashes excaped on windows
         println(f)
     end
 end
 
-@static if (VERSION >= v"0.7.0-DEV.3382" && Sys.isapple()) || (VERSION < v"0.7.0-DEV.3382" && is_apple())
-    Libdl.dlopen("libstdc++",Libdl.RTLD_GLOBAL)
+@static if Sys.isapple()
+    Libdl.dlopen("libstdc++", Libdl.RTLD_GLOBAL)
 end
 
 base_env = "CPLEX_STUDIO_BINARIES"
 
+cplex_path = try
+    # Find the path to the CPLEX executable.
+    @static if Sys.isapple() || Sys.isunix()
+        read(`which cplex`, String)
+    elseif Sys.iswindows()
+        read(`where cplex`, String)
+    end
+catch
+    nothing
+end
+if cplex_path !== nothing
+    # Extract the path to the folder containing the CPLEX executable.
+    cplex_path = dirname(strip(cplex_path))
+end
+
 const cpxvers = ["128", "1280", "129", "1290"]
 
 libnames = String["cplex"]
 for v in reverse(cpxvers)
-    if (VERSION >= v"0.7.0-DEV.3382" && Sys.isapple()) || (VERSION < v"0.7.0-DEV.3382" && is_apple())
+    if Sys.isapple()
         push!(libnames, "libcplex$v.dylib")
-    elseif (VERSION >= v"0.7.0-DEV.3382" && Sys.isunix()) || (VERSION < v"0.7.0-DEV.3382" && is_unix())
+    elseif Sys.isunix()
         push!(libnames, "libcplex$v.so")
         if haskey(ENV, base_env)
             push!(libnames, joinpath(ENV[base_env], "libcplex$v.so"))
         end
+        if cplex_path !== nothing
+            push!(libnames, joinpath(cplex_path, "libcplex$v.so"))
+        end
     end
 end
 
 const wincpxvers = ["128", "1280", "129", "1290"]
-@static if (VERSION >= v"0.7.0-DEV.3382" && Sys.iswindows()) || (VERSION < v"0.7.0-DEV.3382" && is_windows())
+@static if Sys.iswindows()
     for v in reverse(wincpxvers)
         env = base_env * v
-        if haskey(ENV,env)
-            for d in split(ENV[env],';')
+        if haskey(ENV, env)
+            for d in split(ENV[env], ';')
                 occursin("cplex", d) || continue
                 if length(v) == 3 # annoying inconsistency
-                    push!(libnames,joinpath(d,"cplex$(v)0"))
+                    push!(libnames, joinpath(d, "cplex$(v)0"))
                 else
-                    push!(libnames,joinpath(d,"cplex$(v)"))
+                    push!(libnames, joinpath(d, "cplex$(v)"))
                 end
             end
         end
+        if cplex_path !== nothing
+            push!(libnames, joinpath(cplex_path, "cplex$(v)"))
+            push!(libnames, joinpath(cplex_path, "cplex$(v)0"))
```

Comment:
Why adding the `0` ? It is already added in the `wincpxvers` array, isn't it ?

Reply:
Actually, I duplicated what is done above, when checking with the environment variable.

Reply:
Yes, but to be sure that zeros are in the `wincpxvers` array, I would do
```julia
base_wincpxvers = ["128", "129"]
wincpxvers = [base_wincpxvers; base_wincpxvers .* "0"]
```
and same for `cpxvers` for consistency

Reply:
I just did this. How about having only one array instead of two (wincpxvers and cpxvers)?

Reply:
No idea why there are two, they used to be different but I don't know whether we might need two different list again in the future.